etc/vue.api.md

```diff
@@ -18413,6 +18046,10 @@ export interface VueLike {
     focusTarget?: VueLike | Element | Array<VueLike | Element> | null;
 }
 
+// Warnings were encountered during analysis:
+//
+// src/components/FNavigationMenu/FNavigationMenu.vue:530:12 - (ae-forgotten-export) The symbol "MenuItem" needs to be exported by the entry point index.d.ts
```

Comment:
Vid export av typen: `The symbol "items" is marked as @public, but its signature references "MenuItem" which is marked as @internal`, m.fl.

Innebär det att om jag exporterar denna kommer jag även att behöva sätta `@internal` på alla metoder i komponenten som använder typen?

Reply:
Det innebär att konsumenten som försöker använda `items` (propen?) inte kan hänvisa till typen:

```vue
<script setup>
import { FNavigationMenu } from "@fkui/vue";

const items = ref([]);
//    ^^^^^ vad ska items vara för typ? MenuItem är @intern så konsumenten kan inte komma åt den
</script>

<template>
    <f-navigation-menu :items></f-navigation-menu>
    <!--               ^^^^^^ även här blir svårt med type-check för vi vet inte om items är kompatibel eller inte -->
</template>
```

Reply:
Enbart interna metoder i `FNavigationMenu` och utils som använder `MenuItem` nu (kommer från f.d. `IMenu`) så jag antar att konsument inte bör vara intresserad av typen? `NavigationMenuItem` används för prop `routes` istället. Dock liten skillnad mellan dem så jag bör kanske se över om de kan kombineras.

Reply:
Här märkt att options api läcker ut väldigt mycket interna grejer, antingen så ignorerar vi detta om det inte påverkar konsumenten (med risk för att vi senare har en prop eller annat som ska använda typen och att vi då missar det) eller så får vi tyvärr exportera den.

Baserat på vad du säger tycker jag det är ok att ignorera.